lib/modelinstance.js

```diff
@@ -212,7 +212,9 @@ function _encodeValue(context, type, value, forceTyping, f) {
     if (!(value instanceof ModelInstance)) {
       throw new Error('Expected ' + f.name + ' type to be a ModelInstance.');
     }
-    if (type.name !== value.$.schema.name) {
+    // Values must match stated type names, unless the reference is to
+    // 'Mixed', then any reference will do.
+    if (type.name !== value.$.schema.name && (type.name !== 'Mixed')) {
```

Comment:
Interesting.  That sounds like a bug at a higher level of ottoman if the high-level type definition is making it down here.